crypto/src/mls/credential/trust_anchor.rs

```diff
@@ -0,0 +1,1647 @@
+            .collect::<Result<_, x509_cert::der::Error>>()?;
+        Ok(encoded_chain)
+    }
+}
+
+/// Checks the root cert against the trust store. In wasm maybe use webpki-roots (https://github.com/rustls/webpki-roots) crate
+fn check_root_in_trust_store(_root: &Certificate) -> CryptoResult<()> {
+    // TODO: verify root certificate is valid in the device's trust store
+    Ok(())
+}
+
+impl MlsConversation {
+    /// Validates the certificate chain
+    pub(crate) fn validate_received_trust_anchors(
+        old_group_context: &GroupContext,
+        commit_group_context: &GroupContext,
+        backend: &MlsCryptoProvider,
+    ) -> CryptoResult<()> {
+        if let Some(anchors) = commit_group_context.extensions().per_domain_trust_anchors() {
+            // find new anchors
+            anchors
+                .iter()
+                .filter(|anchor| {
+                    old_group_context.extensions().per_domain_trust_anchors().is_none()
+                        || old_group_context
+                            .extensions()
+                            .per_domain_trust_anchors()
+                            .is_some_and(|old_anchors| {
+                                !old_anchors
+                                    .iter()
+                                    .any(|old_anchor| old_anchor.certificate_chain() == anchor.certificate_chain())
+                            })
+                })
+                .try_for_each(|anchor| -> CryptoResult<()> {
+                    let anchor = PerDomainTrustAnchor::try_from(anchor)?;
+                    // the domain will obviously be in the context and will be already applied by
+                    // other client, so the domain uniquennes should not be validated here
+                    anchor.validate(backend, None)?;
+                    Ok(())
+                })?;
+        }
+        Ok(())
+    }
+
+    /// see [MlsCentral::update_trust_anchors_from_conversation]
+    #[cfg_attr(test, crate::durable)]
+    pub(crate) async fn update_trust_anchors(
+        &mut self,
+        client: &Client,
+        remove_domain_names: Vec<String>,
+        add_trust_anchors: Vec<PerDomainTrustAnchor>,
+        backend: &MlsCryptoProvider,
+    ) -> CryptoResult<MlsCommitBundle> {
+        if remove_domain_names.is_empty() && add_trust_anchors.is_empty() {
+            return Err(CryptoError::EmptyTrustAnchorUdpate);
+        }
+
+        let context = self.group.export_group_context();
+        let mut extensions = context.extensions().clone();
+        let mut anchors = extensions
+            .per_domain_trust_anchors()
+            .map(|anchors| {
+                anchors
+                    .iter()
+                    .map(PerDomainTrustAnchor::try_from)
+                    .collect::<CryptoResult<Vec<_>>>()
+            })
+            .unwrap_or_else(|| Ok(Vec::new()))?;
+
+        // check if all to remove exists
+        if remove_domain_names
```

Comment:
You could simplify this by comparing `anchors` length before & after `retain` on L212